### PR TITLE
26 program robot to put ball in storage mechanism

### PR DESCRIPTION
**Pull request by yech199**
Added functions for collecting and releasing balls
- A bit of code cleanup :sweat_smile: 
- **Java**: Use MultipleMotors object instead of arrays of MotorRequest instances
- **Go**: Made motorRequest arrays dynamic

### CODE REVIEW

File: remote/src/main/java/RobotController.java [hunk 1]
```diff
@@ -52,28 +53,76 @@ public void driveStraight(double distance) throws RuntimeException {
...
+    }
+
+    /**
+     * Either collects or releases balls depending on the boolean parameter given
+     * @param isCollecting collects if true and releases if false
+     */
+    public void collectRelease(boolean isCollecting) {
+        MultipleMotors motorRequests;
+        if (isCollecting) {
+            int motorSpeed = -1200;
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:80]: Please comment, use constants on why values are important and not to be changed.

File: remote/src/main/java/RobotController.java [hunk 2]
```diff
@@ -52,28 +53,76 @@ public void driveStraight(double distance) throws RuntimeException {
...
+            motorRequests = createMultipleMotorRequest(Type.m, new MotorPair(OutPort.B, motorSpeed), new MotorPair(OutPort.C, motorSpeed));
+        }
+        else {
+            int speedFrontMotor = 1200;
+            int speedSideMotors = 300;
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:85]: Please comment, use constants on why values are important and not to be changed.

File: remote/src/main/java/RobotController.java [hunk 3]
```diff
@@ -52,28 +53,76 @@ public void driveStraight(double distance) throws RuntimeException {
...
+        }
+
+        CLIENT.collectRelease(motorRequests);
+    }
 
+    /**
+     * Stops the motors collecting/releasing the balls, which currently is port B and C
+     */
+    public void stopCollectRelease() {
+        int motorSpeed = 0;
```
Comment (MikaelFangel) [inline remote/src/main/java/RobotController.java:96]: Java initializes variables to zero per default.

File: robot/main.go [hunk 4]
```diff
@@ -201,19 +218,40 @@ func (s *motorServer) Drive(_ context.Context, in *pBuff.DriveRequest) (*pBuff.S
...
+	for i, request := range in.GetMotor() {
+		motor, err := getMotorHandle(request.GetMotorPort().String(), request.GetMotorType().String())
+		if err != nil {
+			return &pBuff.StatusReply{ReplyMessage: false}, err
+		}
+		if i%2 == 0 { // Currently port B
+			motor.SetSpeedSetpoint(int(request.GetMotorSpeed()))
+		} else { // Currently port C
+			motor.SetSpeedSetpoint(int(-request.GetMotorSpeed()))
+		}
```
Comment (MikaelFangel) [inline robot/main.go:239]: Should be on port names and not on the iterator value.

**Review by MikaelFangel — APPROVED** [review]
Looks fine! (:

File: remote/src/main/java/RobotController.java (hunk 3 above)
Comment (yech199) [reply remote/src/main/java/RobotController.java:96]: If I don't initialize it, I can't pass it down as argument when creating the MotorPair objects

**Pull request merged**